waspc/data/Generator/templates/sdk/wasp/server/_types/index.ts

```diff
@@ -88,20 +87,5 @@ type Context<Entities extends _Entity[]> = Expand<{
 {=# isAuthEnabled =}
 type ContextWithUser<Entities extends _Entity[]> = Expand<Context<Entities> & { user?: AuthUser }>
 
-// TODO: This type must match the logic in auth/session.js (if we remove the
```

Comment:
This TODO is now resolved yay! Since the type and the runtime value are based on the same logic.